python/runtime/cudaq/algorithms/py_observe.cpp

```diff
@@ -23,6 +23,12 @@ constexpr int defaultShotsValue = -1;
 /// @brief Default qpu id value set to 0
 constexpr int defaultQpuIdValue = 0;
 
+/// @brief For asynchronous execution, we need to construct OpaqueArguments 
+/// outside of the async lambda invocation. If we don't, then we will be 
+/// using Python types outside of the current GIL context. Bad things happen then.
+std::unordered_map<std::size_t, std::unique_ptr<OpaqueArguments>>
+    asyncArgsHolder;
```

Comment:
Does the information in this map ever get released?

Reply:
Correct me if I'm wrong, but since this is at global scope, the destructor on the map should get called at program exit. The map destructor should call the unique_ptr destructors, thus releasing the memory.

Reply:
I believe that is correct.

I was wondering if the map got cleaned up when one of the async threads returned.